akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -175,6 +178,42 @@ private[stream] object GraphInterpreter {
         outOwners.mkString("[", ",", "]") +
         ")"
   }
+
+  object GraphAssembly {
+    /**
+     * INTERNAL API
+     */
+    final def apply(inlets: immutable.Seq[Inlet[_]],
+                    outlets: immutable.Seq[Outlet[_]],
+                    stages: GraphStageWithMaterializedValue[_, _]*): GraphAssembly = {
+      // add the contents of an iterator to an array starting at idx
+      @tailrec def add[T](i: Iterator[T], a: Array[T], idx: Int): Array[T] =
+        if (i.hasNext) {
+          a(idx) = i.next()
+          add(i, a, idx + 1)
+        } else a
+
+      // fill array slots with Boundary
+      def markBoundary(owners: Array[Int], from: Int, to: Int): Array[Int] = {
+        Arrays.fill(owners, from, to, Boundary)
```

Comment:
Yes, sadly Arrays.fill does not return the Array, otherwise this method would not really be needed.